deltacat/compute/compactor/steps/repartition.py

```diff
@@ -0,0 +1,162 @@
+            )
+            # check if the column exists in the table
+            if not all(column in table.column_names for table in tables):
+                raise ValueError(f"Column {column} does not exist in the table")
+            # given a range [x, y, z], we need to split the table into 4 files, i.e., (-inf, x], (x, y], (y, z], (z, inf)
+            partitioned_tables_list = [[] for _ in range(len(partition_ranges) + 1)]
+            total_record_count = 0
+            col_name_int64 = f"{column}_int64"
+            for table in tables:
+                total_record_count += len(table)
+                table_new = table.add_column(
+                    0,
+                    pa.field(col_name_int64, pa.int64()),
+                    pc.cast(table[column], pa.int64()),
+                )
+                # handle the partition for values less than or equal to the smallest value
+                partitioned_tables_list[0].append(
+                    table_new.filter(
+                        pc.field(col_name_int64) <= pc.scalar(partition_ranges[0])
+                    )
+                )
+                # Iterate over pairs of values in partition_ranges
+                for i, (lower_limit, upper_limit) in enumerate(
+                    zip(partition_ranges[:-1], partition_ranges[1:]), start=1
+                ):
+                    # Add the table filtered by the lower and upper limits to partitioned_tables
+                    partitioned_tables_list[i].append(
+                        table_new.filter(
+                            (pc.field(col_name_int64) > pc.scalar(lower_limit))
+                            & (pc.field(col_name_int64) <= pc.scalar(upper_limit))
+                        )
+                    )
+                # handle the partition for values greater than the largest value
+                partitioned_tables_list[-1].append(
+                    table_new.filter(
+                        pc.field(col_name_int64) > pc.scalar(partition_ranges[-1])
+                    )
+                )
+
+            # TODO(rootliu) set optimal or max number of records per file to defer the performance degradation due to too many small files
+            partition_table_length = 0
+            # After re-grouping the tables by specified ranges, for each group, we need concat and stage the tables
+            partition_deltas: List[Delta] = []
+            for partition_tables in partitioned_tables_list:
+                if len(partition_tables) > 0:
+                    partition_table = pa.concat_tables(partition_tables)
+                    if len(partition_table) > 0:
+                        partition_table_length += len(partition_table)
+                        partition_delta = deltacat_storage.stage_delta(
+                            partition_table,
+                            destination_partition,
+                            content_type=repartitioned_file_content_type,
+                        )
```

Comment:
The `max_records_per_entry` arg is being passed as None. Can you check if this is creating only 1 file per range?

Reply:
This actually solved the small file size issue. thx